PYME/LMVis/pipeline.py

```diff
@@ -759,63 +762,49 @@ def OpenFile(self, filename= '', ds = None, **kwargs):
                 # This won't effect local file loading even if loading is lazy (i.e. shouldn't cause a regression)
                 ds = self._ds_from_file(fn, **kwargs)
 
-            
-        #wrap the data source with a mapping so we can fiddle with things
-        #e.g. combining z position and focus 
-        mapped_ds = tabular.MappingFilter(ds)
-
-        
-        if 'PixelSize' in kwargs.keys():
-            mapped_ds.addVariable('pixelSize', kwargs['PixelSize'])
-            mapped_ds.setMapping('x', 'x*pixelSize')
-            mapped_ds.setMapping('y', 'y*pixelSize')
-
-        #extract information from any events
-        self.ev_mappings, self.eventCharts = _processEvents(mapped_ds, self.events, self.mdh)
+        # skip the MappingFilter wrapping, etc. in self.addDataSource and add this datasource as-is
+        ds.mdh, ds.events = self.mdh, self.events
+        self.dataSources['results_source'] = ds
```

Comment:
Ideally everything should be routed through `pipeline.addDataSource()` rather than just injecting. Injecting is a bad pattern as it leads to things which won't translate to re-runable recipes. At the moment `pipeline.dataSources` is just an alias for `pipeline.recipe.namespace` but it might become a read-only proxy in the future. It's hard to completely avoid injection during file loading, but it's nice to try.

Reply:
I was trying to avoid a non-recipe MappingFilter on the live-update httptabular in another PR (which I suspect will involve some discussion! Happy to use addDataSource for now and kick that conversation to later)

Reply:
In addition to injection worries, this has the potential to be unsafe with some legacy code which expects the pipeline input to be a mapping filter. May not be a huge issue, and probably fixable with comments for now. Suggest we use the "magic" `'input'` name here to be a 1:1 mapping onto the bakeshop.
```suggestion
        self.dataSources['input'] = ds
```

Reply:
On second thoughts, bakeshop loads data in based on table name - we should probably use `'FitResults` here to match.

```suggestion
        self.dataSources['FitResults'] = ds
```